ctapipe_io_lst/calibration.py

```diff
@@ -240,7 +255,12 @@ def calibrate(self, event: ArrayEventContainer):
 
         for tel_id in event.r0.tel:
             r1 = event.r1.tel[tel_id]
-            waveform = r1.waveform
+            # if `apply_drs4_corrections` is False, we did not fill in the
+            # waveform yet.
+            if r1.waveform is None:
+                waveform = event.r0.tel[tel_id].waveform.astype(np.float32)
```

Comment:
I'm wondering about this line, since it was not there in the previous code. Did you found an actual example when this waveform was missing in r1? Nevertheless, the line looks perfectly safe for me.

Reply:
Before, there was no option to not do any DRS4 corrections before applying this calibration and the R1 would always be filled by the `apply_drs4_corrections` function. No we have the option to completely disable drs4 corrections, and in that case, r1 will not yet be filled at this point.